### PR TITLE
Implement toDot Method for String Representation of VerkleTrie

### PR DESCRIPTION
**Pull request by neotheprogramist**
fixes https://github.com/visoftsolutions/besu-verkle-trie/issues/2

### CODE REVIEW

File: src/main/java/org/hyperledger/besu/ethereum/trie/verkle/node/StemNode.java [hunk 1]
```diff
@@ -297,4 +297,27 @@ public String print() {
   private Bytes extractStem(final Bytes stemValue) {
     return stemValue.slice(0, 31);
   }
+
+
+  @Override
+  public String toDot(Boolean showRepeatingEdges) {
+    StringBuilder result = new StringBuilder()
```
Comment (matkt) [inline src/main/java/org/hyperledger/besu/ethereum/trie/verkle/node/StemNode.java:304]: I will prefer maybe something like that to simplify when we try to read

```java

String result = String.format(
    "%s%s[location=\"%s\", stem=\"%s\", leftCommitment=\"%s\", rightCommitment=\"%s\"]\n",
    getClass().getSimpleName(),
    getLocation().orElse(Bytes.EMPTY),
    getLocation().orElse(Bytes.EMPTY),
    getStem(),
    getLeftCommitment().orElse(Bytes32.ZERO),
    getRightCommitment().orElse(Bytes32.ZERO)
);
```

Comment (matkt) [reply src/main/java/org/hyperledger/besu/ethereum/trie/verkle/node/StemNode.java:304]: the same for the other print

File: src/test/java/org/hyperledger/besu/ethereum/trie/verkle/SimpleVerkleTrieTest.java [hunk 2]
```diff
@@ -25,6 +25,55 @@
...
+    SimpleVerkleTrie<Bytes32, Bytes32> trie = new SimpleVerkleTrie<Bytes32, Bytes32>();
+    Bytes32 key1 = Bytes32.fromHexString("0x00112233445566778899aabbccddeeff00112233445566778899aabbccddeeff");
+    Bytes32 value1 = Bytes32.fromHexString("0x1000000000000000000000000000000000000000000000000000000000000000");
+    Bytes32 key2 = Bytes32.fromHexString("0x00112233445566778899aabbccddeeff00112233445566778899aabbccddee00");
+    Bytes32 value2 = Bytes32.fromHexString("0x0100000000000000000000000000000000000000000000000000000000000000");
+
+    trie.put(key1, value1);
+    trie.put(key2, value2);
+
+    System.out.println(trie.toDotTree());
```
Comment (matkt) [inline src/test/java/org/hyperledger/besu/ethereum/trie/verkle/SimpleVerkleTrieTest.java:51]: I think we are missing assert for the tests

And I will maybe move this meeting to another class in order to not mix the different kind of tests (trie logic and dot display)

**Review by matkt — CHANGES_REQUESTED** [review]

File: src/test/java/org/hyperledger/besu/ethereum/trie/verkle/SimpleVerkleTrieTest.java (hunk 2 above)
Comment (Uacias) [reply src/test/java/org/hyperledger/besu/ethereum/trie/verkle/SimpleVerkleTrieTest.java]: I have addressed the suggestion and moved the tests to another class, as you recommended. However, I'd like to mention that in the tests, there are now very long single-line strings. This is due to the fact that the expected trees can be quite large, and it's a result of Nodes (such as InternalNode, BranchNode, StemNode) to have NullNodes or NullLeafNodes in their children lists.

Comment (matkt) [reply src/test/java/org/hyperledger/besu/ethereum/trie/verkle/SimpleVerkleTrieTest.java]: I was more thinking to put the string in a ressource file and load it in the test to compare with the result 

you can load a file like that in the test 

```java
private String getResources(final String fileName) throws IOException {
    var classLoader = DotDisplayTest.class.getClassLoader();
    return new String(
        classLoader.getResourceAsStream(fileName).readAllBytes(), StandardCharsets.UTF_8);
  }
```

and put your file that contains a string in "src/test/resources/"

**Review by matkt — APPROVED** [review]
LGTM, you need to run spotless cmd because the pipeline is failing

Is it the completly same format as geth ?

**Review by matkt — CHANGES_REQUESTED** [review]
I think we can merge the second one directly as I added comment in the other ?

**Pull request closed without merging**